CDP4Addin/Addin.cs

```diff
@@ -624,6 +632,28 @@ private void AddinOnConnection(object application, ext_ConnectMode connectMode,
             }
         }
 
+        /// <summary>
+        /// Handles the provided exception by showing it to the end-user
+        /// </summary>
+        /// <param name="ex">
+        /// The exception that is being handled
+        /// </param>
+        private static void HandleException(Exception ex)
+        {
+            logger.Error(ex);
+
+            var thread = new Thread(() =>
+            {
+                var exceptionReporter = new ExceptionReporter();
+                exceptionReporter.Show(ex);
+
+                Environment.Exit(1);
```

Comment:
Isn't this a big change? Until now every error was "swallowed" and execution continued. Environment.Exit(1) closes the plugin/Excel right? Does it allow the user to save changes to open Excel documents before closing?

Reply:
Correct the plugin will be closed. But this action is triggered by the user after he focus the error report window(which doesn't act as a modal window) and click close.
Also the user is able to see and save the error.
And because the error window is started in the background the excel application keeps running and is intractable, until is closed.

Reply:
Yes, until now every error was "swallowed" and execution continued (potentially leaving the process in an undefined state).

`Environment.Exit(1)` does indeed close the Excel process.

Due to the fact that the ExceptionReporter is run on a different thread, the user can still interact with Excel until the exception reporter is closed. To me the fact that the exception reporter lets you still interact with Excel did non seem like a desired behavior, but we couldn't get it to run any other way; I guess we could consider this a feature...

Reply:
@cozminvelciu @adrianchivu please remove the `Environment.Exit(1)` for now. Crashing Excel for now is not necessary, at least this was necessary before this change. What is important for us is the error message to analyse.